File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/github/GithubRepository.java
package ru.avystavkin.githubgists.repository.github;

import java.util.List;

import androidx.annotation.NonNull;
import io.reactivex.Observable;
import ru.avystavkin.githubgists.models.local.Gist;
import ru.avystavkin.githubgists.models.local.GistCommit;
import ru.avystavkin.githubgists.models.local.User;

public interface GithubRepository {

    @NonNull
    Observable<List<Gist>> getGists();

    @NonNull
    Observable<List<User>> getPopularUsers(int count);

    @NonNull
    Observable<Gist> getGistById(String id);

    @NonNull
    Observable<List<Gist>> getGistsByUserName(String name);

    @NonNull
    Observable<List<GistCommit>> getCommitsByGistId(String id);

}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/github/GithubRepository.kt
package ru.avystavkin.githubgists.repository.github

import io.reactivex.Observable
import ru.avystavkin.githubgists.models.local.Gist
import ru.avystavkin.githubgists.models.local.GistCommit
import ru.avystavkin.githubgists.models.local.User

interface GithubRepository {

    val gists: Observable<List<Gist>>

    fun getPopularUsers(count: Int): Observable<List<User>>

    fun getGistById(id: String): Observable<Gist>

    fun getGistsByUserName(name: String): Observable<List<Gist>>

    fun getCommitsByGistId(id: String): Observable<List<GistCommit>>

}
